test/controllers/works_controller_test.rb

```diff
@@ -1,116 +1,468 @@
 require 'test_helper'
 
 describe WorksController do
+  let (:movie) {works(:movie)}
+
   describe "root" do
     it "succeeds with all media types" do
       # Precondition: there is at least one media of each category
+      get root_path
 
+      must_respond_with :success
     end
 
     it "succeeds with one media type absent" do
       # Precondition: there is at least one media in two of the categories
+      movie.category = "book"
+
+      get root_path
+
+      must_respond_with :success
 
     end
 
     it "succeeds with no media" do
+      Work.destroy_all
 
+      get root_path
+
+      must_respond_with :success
+      expect(Work.all.count).must_equal 0
     end
   end
 
   CATEGORIES = %w(albums books movies)
   INVALID_CATEGORIES = ["nope", "42", "", "  ", "albumstrailingtext"]
 
-  describe "index" do
-    it "succeeds when there are works" do
-
+  describe "logged in user access" do
+    before do
```

Comment:
This `before` block is causing the CSRF errors/warning.  Instead you should put the `perform_login` inside each `it` block.